### PR TITLE
Fixed incorrect Quartic roots

### LINKED ISSUE #17563: incorrect roots of quartic polynomials
I'm having incorrect roots of generic quartic polynomials. Given this code : 
```
x = symbols('x')

expr1 = Rational(1/2)*x**4 + x**2 - 1 # quartic
res1 = simplify(solveset(expr1, x)) # gives {-I*sqrt(1 + sqrt(3)), -sqrt(-1 + sqrt(3)), sqrt(-1 + sqrt(3)), sqrt(-sqrt(3) - 1)}
assert simplify(expr1.subs(x,res1.args[0])) == 0 # the first root is correct

c = symbols('c[0:5]')
expr2 = c[0]*x**4 + c[1]*x**3 + c[2]*x**2 + c[3]*x + c[4] # generic quartic
roots = solveset(expr2, x) # finding the generic roots
res2 = simplify(roots.subs({c[0]:Rational(1/2), c[1]:0, c[2]:1, c[3]:0, c[4]:-1})) # => substituting coefficients gives a long expression
res2.evalf() # => gives wrong roots : {-1.0⋅ⅈ, 1.0⋅ⅈ}
```
`res1` is correct while `res2` is not.

### PR DESCRIPTION
**Pull request by harsh9200**
#### References to other Issues or PRs

Fixes #17563

#### Release Notes

- polys 
    * add condition for f == 0 to Piecewise solution for symbolic quartic polynomials

### CODE REVIEW

File: sympy/polys/tests/test_polyroots.py [hunk 1]
```diff
@@ -682,3 +682,8 @@ def test_nroots2():
 
 def test_roots_composite():
     assert len(roots(Poly(y**3 + y**2*sqrt(x) + y + x, y, composite=True))) == 3
+
+
+def test_issue_17563():
+    assert simplify(solveset(a*x**4 + b*x**3 + c*x**2 + d*x + e, x).subs({a:Rational(1/2), b:0, c:1, d:0, e:-1})) == \
```
Comment (smichr) [inline sympy/polys/tests/test_polyroots.py]: please wrap lines so none are longer than 72 characters

File: sympy/polys/polyroots.py [hunk 2]
```diff
@@ -355,9 +354,12 @@ def _ans(y):
             if fuzzy_not(p.is_zero):
                 return _ans(y2)
 
+            _y1, _y2 = [sqrt(tmp) for tmp in
+                      roots([1, e, g], multiple=True)]
+
             # sort it out once they know the values of the coefficients
-            return [Piecewise((a1, Eq(p, 0)), (a2, True))
-                for a1, a2 in zip(_ans(y1), _ans(y2))]
+            return [Piecewise((a1, Eq(p, 0)),(a3 - aon4, Eq(f, 0)), (a2, True))
```
Comment (smichr) [inline sympy/polys/polyroots.py]:
```suggestion
            return [Piecewise((a1, Eq(p, 0)), (a3 - aon4, Eq(f, 0)), (a2, True))
```

File: sympy/polys/tests/test_polyroots.py (hunk 1 above)
Comment (harsh9200) [reply sympy/polys/tests/test_polyroots.py]: okay doing

**Comment by smichr** [comment]
Please squash the commits into a single one (the first one).

**Comment by harsh9200** [comment]
@smichr Done

**Pull request merged**